### PR TITLE
Added names and helper classes to joint base and moteus

### PR DESCRIPTION
**Pull request by jdcaporale**
This PR 
- adds name member to the JointBase and JointMoteus classes and constructors
- fixes a small bug in the joint base default for pos_max
- adds the JointSharedVector class
- adds the MoteusJointConfig struct for simpler construction
- and fixes some formatting and commenting errors

### CODE REVIEW

**Review by ethanmusser — APPROVED** [review]
LGTM.  Adds much-desired joint names and support for using them.  These can be used to implement joint name-to-id mapping in a future PR in the `JointSharedVector` object.

File: include/kodlab_mjbots_sdk/common_header.h [hunk 1]
```diff
@@ -3,6 +3,9 @@
 #pragma once
 #include <atomic>
 #include <signal.h>  // manage the ctrl+c signal
+#include <type_traits>
```
Comment (jdcaporale) [inline include/kodlab_mjbots_sdk/common_header.h:6]:
```suggestion
```

File: include/kodlab_mjbots_sdk/common_header.h [hunk 2]
```diff
@@ -3,6 +3,9 @@
...
+#include <iostream>
```
Comment (jdcaporale) [inline include/kodlab_mjbots_sdk/common_header.h:7]:
```suggestion
```

File: include/kodlab_mjbots_sdk/common_header.h [hunk 3]
```diff
@@ -3,6 +3,9 @@
...
+#include <vector>
```
Comment (jdcaporale) [inline include/kodlab_mjbots_sdk/common_header.h:8]:
```suggestion
```
This PR is not the place to add these

**Pull request merged**